resolwe_bio/processes/reads_processing/trimmomatic.yml

```diff
@@ -131,193 +149,281 @@
-        {% if slidingwindow.window_size and slidingwindow.required_quality %}"SLIDINGWINDOW:"{{ slidingwindow.window_size }}":"{{ slidingwindow.required_quality }}{% endif %} \
-        {% if maxinfo.target_length and maxinfo.strictness %}}"MAXINFO:"{{maxinfo.target_length}}":"{{maxinfo.strictness}}{% endif %} \
-        {% if leading %}"LEADING:"{{ leading }}{% endif %} \
-        {% if trailing %}"TRAILING:"{{ trailing }}{% endif %} \
-        {% if crop %}"CROP:"{{ crop }}{% endif %} \
-        {% if headcrop %}"HEADCROP:"{{ headcrop }}{% endif %} \
-        {% if minlen %}"MINLEN:"{{minlen}}{% endif %} \
-        {% if average_quality %}"AVGQUAL:"{{ average_quality }}{% endif %}
-
-      re-checkrc "Failed while processing with Trimmomatic."
-      re-progress 0.7
-
-      mv out_reads_filtered.fastq.gz "${NAME}_filtered.fastq.gz"
-
-      echo "Postprocessing FastQC..."
-
-      # http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
-      mkdir "fastqc" && fastqc "${NAME}_filtered.fastq.gz" --extract --outdir="fastqc" >&2
-      re-checkrc "Failed while processing with FastQC."
-      re-progress 0.9
-
-      mv "fastqc/${NAME}_filtered_fastqc" "fastqc/${NAME}_fastqc"
-      mv "fastqc/${NAME}_filtered_fastqc.zip" "${NAME}_fastqc.zip"
-
-      FASTQC_URL="{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}"
-
-      re-progress 0.95
-
-      re-save-file-list fastq "${NAME}_filtered.fastq.gz"
-      re-save-file stats stats.txt
+      {% if
+        not reads_filtering.average_quality
+        and not reads_filtering.minlen
+        and not trim_bases.headcrop
+        and not trim_bases.crop
+        and not trim_bases.trailing
+        and not trim_bases.leading
+        and not maxinfo.target_length
+        and not  maxinfo.strictness
+        and not slidingwindow.window_size
+        and not slidingwindow.required_quality
+        and not illuminaclip.adapters
+        and not illuminaclip.seed_mismatches
+        and not illuminaclip.simple_clip_threshold %}
+        re-warning "The reads are to be copied from the input to the output, including the quality control report.
+      To process reads with Trimmomatic please define at least one parameter."
+
+        cp in_reads.fastq.gz "${NAME}.fastq.gz"
+        cp {{ reads.fastqc_archive.0.file }} "${NAME}_fastqc.zip"
+
+        mkdir "fastqc"
+        cp -r {{ reads.fastqc_url.0.refs.0}} "fastqc/${NAME}_fastqc"
+
+        FASTQC_URL="{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}"
+
+        re-save-file-list fastq "${NAME}.fastq.gz"
+        re-save-file-list fastqc_archive "${NAME}_fastqc.zip"
+        re-save-list fastqc_url "${FASTQC_URL}"
+
+      {% else %}
+
+        echo "Removing adapters and filtering data..."
+
+        {% if (illuminaclip.adapters and not illuminaclip.seed_mismatches)
```

Comment:
What happens, if `seed_mismatches` and `simple_clip_threshold` are specified but `adapters` are not?

Reply:
The condition check is incorrect. If `seed_mismatches` and `simple_clip_threshold` are specified but `adapters` are not the error message will not be displayed and an error will occur in the `else` part of the statement where `adapters` are first accessed.